js/app.js

```diff
@@ -210,6 +165,12 @@ function appendBlogTagToTagCloud(blogTag) {
 }
 
 
+function navigateToBlogs(blogTag) {
+    var urlToNavigate = location.protocol + '//' + location.host + location.pathname;
+    window.location.href = urlToNavigate + (blogTag ? "?tag=" + blogTag : "");
+}
```

Comment:
If I understand the intent, the idea here is:
1. Start with the current URL
1. Remove any querystring (search) or hash tag (fragment) from the URL
1. Append any blog tag which is needed
1. Nagivate to the newly calculated URL

Your implementation will work as long as the blog isn't being served over anything other than the default port (80 or 443). In order to avoid any potential problems, you can use the built in `URL` API like this:

```javascript
function navigateToBlogs(blogTag) {
    const baseURL = location.href; // this will be used to retreive protocol, port, host etc.
    const url = new URL(location.pathname, baseURL); 
    if(blogTag) url.searchParams.set("tag", blogTag);
    location.href = url.toString()
}
```

Reply:
Great, thanks